vm/src/emulator/executor.rs

```diff
@@ -1184,21 +1185,55 @@ impl Emulator for LinearEmulator {
                         value: byte,
                     })
             });
-        // TODO: avoid creating a BtreeMap and produce an iterator directly
-        let rom_initialization = match self.static_rom_image_index {
-            None => BTreeMap::new(),
-            Some(uidx) => self
-                .memory
-                .addr_val_bytes(uidx)
-                .expect("invalid static_rom_image_index"),
-        };
-        let rom_iter = rom_initialization
-            .iter()
-            .map(|(addr, byte)| MemoryInitializationEntry {
-                address: *addr,
-                value: *byte,
-            });
 
+        // Helper function to map (addr, byte) to MemoryInitializationEntry
+        fn map_to_mem_init_entry((addr, byte): (u32, u8)) -> MemoryInitializationEntry {
+            MemoryInitializationEntry {
+                address: addr,
+                value: byte,
+            }
+        }
+        let mut rom_count = 0;
+        let rom_iter = match self.static_rom_image_index {
+            None => std::iter::empty().collect::<Vec<_>>().into_iter(),
+            Some((store, idx)) => {
+                if store == Modes::RW as usize {
```

Comment:
You should be able to do this via pattern matching rather than as an if statement, I think?